### PR TITLE
feat!: expose reason for transaction cancellation for callback in wallet_ffi

### PR DESCRIPTION
**Pull request by StriderDM**
Description
---
Exposes reason for transaction cancellation to wallet_ffi via the transaction cancellation callback.

Additionally removes outdated docs and fixes Clippy errors.

Motivation and Context
---

How Has This Been Tested?
---
cargo test --all

### CODE REVIEW

File: base_layer/wallet/src/transaction_service/handle.rs [hunk 1]
```diff
@@ -150,7 +150,7 @@ pub enum TransactionEvent {
     TransactionDirectSendResult(TxId, bool),
     TransactionCompletedImmediately(TxId),
     TransactionStoreForwardSendResult(TxId, bool),
-    TransactionCancelled(TxId),
+    TransactionCancelled(TxId, u64),
```
Comment (stringhandler) [inline base_layer/wallet/src/transaction_service/handle.rs]: Rather use the actual enum here and cast it to u64 when it crosses an application boundary
```suggestion
    TransactionCancelled(TxId, TxRejection),
```

File: base_layer/wallet_ffi/src/lib.rs [hunk 2]
```diff
@@ -3085,7 +3056,7 @@ pub unsafe extern "C" fn wallet_create(
     callback_transaction_mined_unconfirmed: unsafe extern "C" fn(*mut TariCompletedTransaction, u64),
     callback_direct_send_result: unsafe extern "C" fn(c_ulonglong, bool),
     callback_store_and_forward_send_result: unsafe extern "C" fn(c_ulonglong, bool),
-    callback_transaction_cancellation: unsafe extern "C" fn(*mut TariCompletedTransaction),
+    callback_transaction_cancellation: unsafe extern "C" fn(*mut TariCompletedTransaction, u64),
```
Comment (philipr-za) [inline base_layer/wallet_ffi/src/lib.rs:3074]: Should be documented in the docstring above and in `wallet.h` docstring what this callback does. It should include a description of the enum values.

Comment (StriderDM) [reply base_layer/wallet_ffi/src/lib.rs:3074]: Whoops missed that, thanks @philipr-za

File: base_layer/wallet_ffi/src/lib.rs [hunk 3]
```diff
@@ -3044,9 +3015,22 @@ unsafe fn init_logging(
 /// when a Broadcast transaction is detected as mined AND confirmed.
 /// `callback_transaction_mined_unconfirmed` - The callback function pointer matching the function signature. This will
 /// be called  when a Broadcast transaction is detected as mined but not yet confirmed.
-/// `callback_discovery_process_complete` - The callback function pointer matching the function signature. This will be
-/// called when a `send_transacion(..)` call is made to a peer whose address is not known and a discovery process must
-/// be conducted. The outcome of the discovery process is relayed via this callback
+/// `callback_direct_send_result` - The callback function pointer matching the function signature. This is called
+/// when a direct send is completed. The first parameter is the transaction id and the second is whether if was
+/// successful or not. `callback_store_and_forward_send_result` - The callback function pointer matching the function
```
Comment (philipr-za) [inline base_layer/wallet_ffi/src/lib.rs]: I think a rustfmt screwed up the line breaks in this comment

**Review by philipr-za — APPROVED** [review]

**Pull request merged**